Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,769 @@
+                       GeomTraits());
+
+  std::ptrdiff_t cutoff = 2000;
+  CGAL::box_intersection_d(box1_ptr.begin(), box1_ptr.end(),
+                           box2_ptr.begin(), box2_ptr.end(),
+                           Intersect_faces,cutoff);
+  return Intersect_faces.m_iterator;
+}
+
+
+/*
+ * \ingroup PMP_intersection_grp
+ * detects and records intersections between the specified
+ * faces of a triangulated surface mesh and a `Polyline`.
+ * This function depends on the package \ref PkgBoxIntersectionDSummary
+ *
+ * \pre `CGAL::is_triangle_mesh(mesh)`
+ *
+ * \tparam TriangleMesh a model of `FaceListGraph`
+ * \tparam FaceRange range of `boost::graph_traits<TriangleMesh>::%face_descriptor`,
+ *  model of `Range`.
+ * Its iterator type is `RandomAccessIterator`.
+ * \tparam Polyline a `RandomAccessRange` of `CGAL::Point_3`. Those points must come from the same
+ * geometric traits that templates`TriangleMesh`.
+ * \tparam OutputIterator a model of `OutputIterator` holding objects of type
+ *   `std::pair<std::size_t, std::size_t>`. This OutputIterator will hold the position of the
+ *  elements in their respective range.
+ * \tparam NamedParameters a sequence of \ref namedparameters
+ *
+ * \param face_range the range of `mesh` faces to check for intersections.
+ * \param polyline the `Polyline` to check for intersections.
+ * \param mesh the triangulated surface mesh to be checked.
+ * \param out output iterator to be filled with all pairs of face-segment that intersect
+ * \param np optional sequence of \ref namedparameters for `mesh`, among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`.
+ *   If this parameter is omitted, an internal property map for
+ *   `CGAL::vertex_point_t` should be available in `TriangleMesh`\cgalParamEnd
+ *    \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `PMPSelfIntersectionTraits` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ * \return `out`
+ */
+template <class TriangleMesh
+          , class FaceRange
+          , class Polyline
+          , class OutputIterator
+          , class NamedParameters
+          >
+OutputIterator
+compute_face_polyline_intersection( const FaceRange& face_range,
+               const Polyline& polyline,
+               const TriangleMesh& mesh,
+               OutputIterator out,
+               const NamedParameters& np)
+{
+  CGAL_precondition(CGAL::is_triangle_mesh(mesh));
+
+  typedef TriangleMesh TM;
+  typedef typename boost::graph_traits<TM>::face_descriptor face_descriptor;
+  typedef typename GetVertexPointMap<TM, NamedParameters>::const_type VertexPointMap;
+
+  VertexPointMap vpmap = boost::choose_param(get_param(np, internal_np::vertex_point),
+                                             get_const_property_map(boost::vertex_point, mesh));
+  typedef typename boost::property_traits<VertexPointMap>::value_type Point;
+
```

Comment:
You can add some `CGAL_static_assertion` to be sure that the point types are the same. You can also do the same for the vertex point map for the different meshes. In corefinement I have put:
```
  typedef typename GetVertexPointMap<TriangleMesh,
                                     NamedParameters1>::type Vpm;
  typedef typename GetVertexPointMap<TriangleMesh,
                                     NamedParameters2>::type Vpm2;
  CGAL_USE_TYPE(Vpm2);
  CGAL_assertion_code(
    static const bool same_vpm = (boost::is_same<Vpm,Vpm2>::value); )
```
This will help a lot to have the error in the static assertion rather than a cryptic message later.